OrbitCore/LinuxTracingSession.cpp

```diff
@@ -0,0 +1,94 @@
+#include "LinuxTracingSession.h"
+
+#include <utility>
+
+void LinuxTracingSession::RecordContextSwitch(ContextSwitch&& context_switch) {
+  absl::MutexLock lock(&context_switch_buffer_mutex_);
+  context_swith_buffer_.push_back(context_switch);
+}
+
+void LinuxTracingSession::RecordTimer(Timer&& timer) {
+  absl::MutexLock lock(&timer_buffer_mutex_);
+  timer_buffer_.push_back(timer);
+}
+
+void LinuxTracingSession::RecordCallstack(LinuxCallstackEvent&& event) {
+  absl::MutexLock lock(&callstack_buffer_mutex_);
+  callstack_buffer_.push_back(event);
+}
+
+void LinuxTracingSession::RecordHashedCallstack(
+    CallstackEvent&& hashed_call_stack) {
+  absl::MutexLock lock(&hashed_callstack_buffer_mutex_);
+  hashed_callstack_buffer_.push_back(std::move(hashed_call_stack));
+}
+
+bool LinuxTracingSession::ReadAllContextSwitches(
+    std::vector<ContextSwitch>* buffer) {
+  absl::MutexLock lock(&context_switch_buffer_mutex_);
+  if (context_swith_buffer_.empty()) {
+    return false;
+  }
+
+  std::swap(context_swith_buffer_, *buffer);
```

Comment:
Nit: Maybe a std::move here? Makes it slightly clearer what's going on

Reply:
done